core/modules/set_metadata.cc

```diff
@@ -79,10 +97,10 @@ CommandResponse SetMetadata::AddAttrOne(
   }
   name = attr.name();
   size = attr.size();
+  do_mask = attr.mask().length() > 0;
```

Comment:
It would be useful if this function checks if mask and shift are specified only for "copy from packet/metadata" mode (for "copy from value" mask&shift are pointless)